blocks/library/image/index.js

```diff
@@ -174,7 +176,23 @@ registerBlockType( 'core/image', {
 				</InspectorControls>
 			),
 			<figure key="image" className={ classes }>
-				<img src={ url } alt={ alt } onClick={ setFocus } />
+				<ImageSize src={ url }>
+					{ ( originalWidth = width, originalHeight = height ) => {
+						if ( ! originalHeight || ! originalWidth ) {
+							return <img src={ url } alt={ alt } onClick={ setFocus } />;
+						}
+						return (
+							<ResizableBox
+								width={ originalWidth }
+								height={ originalHeight }
+								lockAspectRatio
+								onResize={ ( event, { size } ) => setAttributes( size ) }
+							>
+								<img src={ url } alt={ alt } onClick={ setFocus } />
```

Comment:
Minor: We could create a single reference to this element to share between condition above and here:

```diff
diff --git a/blocks/library/image/index.js b/blocks/library/image/index.js
index f6985bc8..766fb8aa 100644
--- a/blocks/library/image/index.js
+++ b/blocks/library/image/index.js
@@ -178,8 +178,9 @@ registerBlockType( 'core/image', {
                        <figure key="image" className={ classes }>
                                <ImageSize src={ url }>
                                        { ( originalWidth = width, originalHeight = height ) => {
+                                               const img = <img src={ url } alt={ alt } onClick={ setFocus } />;
                                                if ( ! originalHeight || ! originalWidth ) {
-                                                       return <img src={ url } alt={ alt } onClick={ setFocus } />;
+                                                       return img;
                                                }
                                                return (
                                                        <ResizableBox
@@ -188,7 +189,7 @@ registerBlockType( 'core/image', {
                                                                lockAspectRatio
                                                                onResize={ ( event, { size } ) => setAttributes( size ) }
                                                        >
-                                                               <img src={ url } alt={ alt } onClick={ setFocus } />
+                                                               { img }
                                                        </ResizableBox>
                                                );
                                        } }
```